### PR TITLE
feat: exclude userId from serialized nested objects

### PR DESCRIPTION
**Pull request by mattnichols**
# Summary of Changes

Add a new annotation to exclude the userId field from MdxBase from serializing on classes that are nested.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by gerfboy — DISMISSED** [review]

File: mdx-models/src/main/java/com/mx/path/model/mdx/model/MdxBase.java [hunk 1]
```diff
@@ -37,4 +36,21 @@ public final void appendWarning(Warning warning) {
...
+
+  public final void setUserId(String userId) {
+    if (!INCLUDE_USER_ID_MAP.containsKey(this.getClass())) {
+      IncludeUserId annotation = this.getClass().getAnnotation(IncludeUserId.class);
+      if (annotation != null) {
+        INCLUDE_USER_ID_MAP.put(this.getClass(), annotation.value());
+      } else {
+        INCLUDE_USER_ID_MAP.put(this.getClass(), true);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (mattnichols) [reply mdx-models/src/main/java/com/mx/path/model/mdx/model/MdxBase.java:48]: Yeah. This makes sense. Thanks for the feedback. Changing.

File: mdx-models/src/main/java/com/mx/path/model/mdx/model/MdxBase.java [hunk 2]
```diff
@@ -37,4 +36,18 @@ public final void appendWarning(Warning warning) {
     }
     this.warnings.add(warning);
   }
+
+  public final void setUserId(String userId) {
+    // Don't set the user ID if the model is annotated with MdxNested. Cache result to avoid reflection overhead.
+    if (!INCLUDE_USER_ID_MAP.containsKey(this.getClass())) {
+      MdxNested annotation = this.getClass().getAnnotation(MdxNested.class);
```
Comment (stevecl5) [inline mdx-models/src/main/java/com/mx/path/model/mdx/model/MdxBase.java:43]: Is there a reason `getAnnotation()` is preferred to `isAnnotationPresent()`? It looks like `isAnnotationPresent()` just calls `getAnnotation()` under the hood and does a null check. If the performance is the same, I think `isAnnotationPresent()` is more readable.

**Review by gerfboy — APPROVED** [review]

File: mdx-models/src/main/java/com/mx/path/model/mdx/model/MdxBase.java (hunk 2 above)
Comment (mattnichols) [reply mdx-models/src/main/java/com/mx/path/model/mdx/model/MdxBase.java:43]: I agree. This was just a holdover from the previous implementation. Fixing.

Comment (mattnichols) [reply mdx-models/src/main/java/com/mx/path/model/mdx/model/MdxBase.java]: This is fixed.

**Review by gerfboy — APPROVED** [review]

**Pull request merged**